process/smartContract/process.go

```diff
@@ -971,6 +971,13 @@ func (sc *scProcessor) ProcessIfError(
 		return err
 	}
 
+	txType, _ := sc.txTypeHandler.ComputeTransactionType(tx)
+	isCrossShardMoveBalance := txType == process.MoveBalance && check.IfNil(acntSnd)
+	if isCrossShardMoveBalance {
+		// move balance was already consumed in sender shard
+		return nil
```

Comment:
This change will keep the backwards compatibility ?

Reply:
We can add a flag, only possibility is sending value to SC cross shard

Reply:
done.